### PR TITLE
Feature/lynguyxn/chess draw handler

### LINKED ISSUE #29: Draw Game
## Goals
- Implement the Draw button below the chessboard as designed in figma
- Link button to emitting Draw events as designed in the backend event handlers
- Add listeners for Draw events from the server 
  - Use to render "Your Opponent Would Like a Draw" popup
    - Emit relevant event for Yes/No buttons
  - Use to render "Draw Request Accepted"
    - Navigate back to home 
    - End Game
  - Use to render "Draw Request Declined"

### PR DESCRIPTION
**Pull request by Miyuki-L**
### Administrative Info
[GitHub Issue #29]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added draw button to chess screen
-  Disabled draw button function when button is grayed out
- Handled proposing and receiving draw request in the frontend
- Handled accepting and rejecting draw request in the frontend
- Changed Chess Screen Background to White

### Testing
How did you confirm your changes worked? 
- Tested sending and receiving draw request on two devices
- Tested accepting and rejecting draw request on two devices

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Chess Screen with Draw Button |
| --- |
|  |

| Chess Screen with Grayed Out Draw Button |
| --- |
|  |

| Draw Request Received Popup |
| --- |
|   |

| Draw Request Accepted Popup |
| --- |
|   |

| Draw Request Declined Popup |
| --- |
|   |

### CODE REVIEW

File: mobile/src/components/popups/popup_templates/OneButtonPopup.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { Alert, Modal, Text, Pressable, View } from "react-native";
```
Comment (RyanHUNGry) [inline mobile/src/components/popups/popup_templates/OneButtonPopup.tsx]: In my popup PR, I moved this template one directory up and removed the popup_templates directory so you guys will have to do the sameand all connected import statements to match this change.

File: mobile/src/components/popups/popup_templates/TwoButtonPopup.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { Alert, Modal, Text, Pressable, View } from "react-native";
```
Comment (RyanHUNGry) [inline mobile/src/components/popups/popup_templates/TwoButtonPopup.tsx]: Same thing here with directory change

File: mobile/src/screens/Chess.tsx [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
 // Source Code: https://github.com/wcandillon/can-it-be-done-in-react-native/tree/master/season4/src/Chess
-import React from "react";
+import { useNavigation } from "@react-navigation/native";
+import React, { useContext, useEffect, useState } from "react";
 import { View, StyleSheet } from "react-native";
 import { gestureHandlerRootHOC } from "react-native-gesture-handler";
 
+import Button from "../components/Button";
 import Board from "../components/chess/Board";
+import OneButtonPopup from "../components/popups/popup_templates/OneButtonPopup";
```
Comment (RyanHUNGry) [inline mobile/src/screens/Chess.tsx]: change to "../components/popups/OneButtonPopup"

File: mobile/src/screens/Chess.tsx [hunk 4]
```diff
@@ -1,11 +1,16 @@
...
+import TwoButtonPopup from "../components/popups/popup_templates/TwoButtonPopup";
```
Comment (RyanHUNGry) [inline mobile/src/screens/Chess.tsx]: change to "../components/popups/TwoButtonPopup"

**Review by RyanHUNGry — CHANGES_REQUESTED** [review]

File: backend/src/eventHandlers/drawHandler.ts [hunk 5]
```diff
@@ -9,6 +9,7 @@ module.exports = function ({ socket, io, username, roomsMap, boards }: GameHandl
   // if client B wants to decline the draw, client B should emit "draw rejected"
   //  in this case, client A should handle "draw request rejected" with username of client B being sent
   socket.on("try draw", () => {
+    console.log("proposing draw received");
```
Comment (nicolaswl) [inline backend/src/eventHandlers/drawHandler.ts]: Remove the log statement if it was just used for debugging

File: mobile/src/components/popups/popup_templates/OneButtonPopup.tsx (hunk 1 above)
Comment (lynguyxn) [reply mobile/src/components/popups/popup_templates/OneButtonPopup.tsx]: resolved

**Review by nicolaswl — APPROVED** [review]
LGTM

File: backend/src/eventHandlers/drawHandler.ts (hunk 5 above)
Comment (lynguyxn) [reply backend/src/eventHandlers/drawHandler.ts]: already removed in latest version

**Review by RyanHUNGry — APPROVED** [review]

**Pull request merged**